File: apps/mobile/src/lib/session.ts
/** Session tokens (IDT-02) live in the OS keychain via expo-secure-store. */
import * as SecureStore from 'expo-secure-store';

const ACCESS_KEY = 'swab.session.access.v1';
const REFRESH_KEY = 'swab.session.refresh.v1';

export interface SessionTokens {
  accessToken: string;
  refreshToken: string;
}

export async function saveTokens(tokens: SessionTokens): Promise<void> {
  await Promise.all([
    SecureStore.setItemAsync(ACCESS_KEY, tokens.accessToken),
    SecureStore.setItemAsync(REFRESH_KEY, tokens.refreshToken),
  ]);
}

export async function getAccessToken(): Promise<string | null> {
  return SecureStore.getItemAsync(ACCESS_KEY);
}

File: apps/ios/Sources/SwabCore/Identity/Session.swift
/// Session tokens (IDT-02) live in the OS keychain via `SecureStore`.
import Foundation

public struct SessionTokens: Equatable, Sendable {
    public let accessToken: String
    public let refreshToken: String

    public init(accessToken: String, refreshToken: String) {
        self.accessToken = accessToken
        self.refreshToken = refreshToken
    }
}

public struct Session: Sendable {
    static let accessKey = "swab.session.access.v1"
    static let refreshKey = "swab.session.refresh.v1"

    private let store: SecureStore

    public init(store: SecureStore) {
        self.store = store
    }

    public func saveTokens(_ tokens: SessionTokens) throws {
        try store.set(Self.accessKey, value: tokens.accessToken)
        try store.set(Self.refreshKey, value: tokens.refreshToken)
    }

    public func getAccessToken() throws -> String? {
        try store.get(Self.accessKey)
    }
}
